aesara/link/jax/jax_dispatch.py

```diff
@@ -184,6 +170,45 @@ def jax_func(*inputs):
     return jax_funcs
 
 
+def get_jax_data_func(i_type, i, idx=None):
+    def jax_data_func(*inputs, i=i, idx=idx):
+        i_dtype = getattr(i, "dtype", None)
+        i_data = i.data if idx is None else inputs[idx]
+
+        if i_dtype is None:
+            return i_data
+        else:
+            return jnp.array(i_data, dtype=jnp.dtype(i_dtype))
+
+    return jax_data_func
+
+
+@singledispatch
+def jaxify(input):
+    return input
+
+
+@jaxify.register(list)
+def jaxify_list(input):
+    for i, val in enumerate(input):
+        input[i] = jaxify(val)
+    return input
+
+
+@jaxify.register(dict)
+def jaxify_dict(input):
+    for key in input:
+        input[key] = jaxify(input[key])
+    return input
+
+
+@jaxify.register(RandomState)
+def jaxify_randomstate(input):
+    state = input.get_state()
+    out = jax.random.PRNGKey(state[1][0])
+    return out
+
+
```

Comment:
Ideally, `jaxify` and `get_jax_data_func` would be a _single_ function that dispatches on `Type`s (i.e. `i_type`).  This fits the Aesara graph paradigm very well and makes it very clear what needs to be done when a new `Type` is added.

Also, `jaxify` is a little to general of a name.  With the strictly `Type`-based dispatching mentioned above, we could have a `jax_typify` (or we could change the names of both to something more simple).

Reply:
The main problem here is `jaxify`/`jax_typify` over here is supposed to return objects themselves rather than functions like `get_jax_data_func` would. Though I can make `get_jax_data_func` use `jax_typify` internally.